src/java.base/share/classes/jdk/internal/foreign/abi/LinkerOptions.java

```diff
@@ -63,7 +63,11 @@ private static LinkerOptions forShared(BiConsumer<LinkerOptionImpl, FunctionDesc
             optionMap.put(option.getClass(), opImpl);
         }
 
-        return new LinkerOptions(optionMap);
+        LinkerOptions linkerOptions = new LinkerOptions(optionMap);
+        if (linkerOptions.hasCapturedCallState() && linkerOptions.isTrivial()) {
+            throw new IllegalArgumentException("captureCallState and isTrivial are mutually exclusive");
```

Comment:
The message is good. I wonder if we could get it to scale better with something like "incompatible linker options: <option list>" which is sometimes what we use e.g. in the Java compiler (e.g. for flags). I wonder e.g. if there might be other set of incompatible options we might need to worry about in the future - seems a possibility. (In any case, we can always change the message later, so this is not a big priority).

Reply:
I like that suggestion.